x-pack/plugins/fleet/server/services/agent_policy.ts

```diff
@@ -383,11 +383,16 @@ class AgentPolicyService {
       throw new FleetError(agentPolicySO.error.message);
     }
 
-    const agentPolicy = { id: agentPolicySO.id, ...agentPolicySO.attributes };
+    const agentPolicy = {
+      id: agentPolicySO.id,
+      ...agentPolicySO.attributes,
+    };
 
     if (withPackagePolicies) {
       agentPolicy.package_policies =
         (await packagePolicyService.findAllForAgentPolicy(soClient, id)) || [];
+
+      await this.populateRootIntegrations?.(soClient, agentPolicy as AgentPolicy);
```

Comment:
is there a reason for adding this as a separate `root_integrations` property vs putting it with `package_policies`, for example under something like `package.requires_root: boolean`?

Reply:
+1 I think it will be a good idea to persist it on the package policy,  `getPackageInfo` is a sensitive code path it call the registry to get the latest package.

Reply:
I'll take a look, initially I saw that this info is not in `package_policies`, and thought it's enough to populate when querying one agent policy.
It would make sense to persist on the package policy when creating it (or update if a package sets root on a new version).

Reply:
refactored to store `package.requires_root` in `ingest-package-policies` SO